### PR TITLE
Respect 'showDashboard' disabled by default

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Fixes https://github.com/microsoft/azuredatastudio/issues/22869

`ShowDashboard` should be not enabled by default for any connection type, as it fails its purpose if connection type is not defined. This PR enables 'showDashboard' where it should be visible, and retains the public API promise to not show dashboard unless specified.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -268,6 +268,23 @@
 				"group": "2_attach"
 			}
 		},
+		{
+			"type": "pwa-chrome",
+			"request": "attach",
+			"name": "Attach to VS Code",
```
Comment (kburtram) [inline .vscode/launch.json:274]: You might has a more descriptive task name related to ADS, such as "Attach to render process", etc.  Though I thought we already had tasks for all the processes.

File: src/sql/workbench/contrib/commandLine/electron-browser/commandLine.ts [hunk 2]
```diff
@@ -234,7 +239,13 @@ export class CommandLineWorkbenchContribution implements IWorkbenchContribution,
 			}
 
 			const connectionProfile = this.readProfileFromArgs(args);
-			await this._connectionManagementService.showConnectionDialog(undefined, undefined, connectionProfile);
+			await this._connectionManagementService.showConnectionDialog(undefined, {
+				saveTheConnection: true,
+				showDashboard: true,
+				params: undefined,
```
Comment (kburtram) [inline src/sql/workbench/contrib/commandLine/electron-browser/commandLine.ts:245]: sometimes we're including `params = undefined` and other call we don't.  Any reason why?

**Review by kburtram — APPROVED** [review]

**Review by Charles-Gagnon — APPROVED** [review]

**Review by alanrenmsft — APPROVED** [review]

File: src/sql/workbench/contrib/commandLine/electron-browser/commandLine.ts (hunk 2 above)
Comment (cheenamalhotra) [reply src/sql/workbench/contrib/commandLine/electron-browser/commandLine.ts:245]: They would mean the same thing, I can remove from all spots to keep it consistent.

File: .vscode/launch.json (hunk 1 above)
Comment (Charles-Gagnon) [reply .vscode/launch.json:274]: Can you do this in a separate PR to avoid multiple unrelated changes being made in a single PR?

**Pull request merged**